### PR TITLE
install composer

### PR DESCRIPTION
**Pull request by Preovaleo**
HI,

i make a composer.json file
and this solve #3471

what to do next:
- [ ]  enable the package on packagist (i can do that too)

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  "author": {
+    "url": "http://agafonkin.com/en/",
+    "name": "Vladimir Agafonkin"
+  },
+  "support": {
+    "issues": "https://github.com/Leaflet/Leaflet/issues",
+    "forum": "https://stackoverflow.com/questions/tagged/leaflet",
+    "source": "https://github.com/Leaflet/Leaflet",
+    "docs": "http://leafletjs.com/reference.html",
+    "extra": {
```
Comment (Preovaleo) [inline composer.json]: not in support

**Comment by mourner** [comment]
Yeah, we don't want to support every possible package manager out there and have hundreds of files like this in the repo root.

**Comment by pvhee** [comment]
@mourner I'd argue against this saying `composer` isn't just every other package manager, but widely used within the PHP community, and other PHP-dependant communities like Drupal / Symfony / etc. are relying increasingly on composer.

Leaflet is widely used within the PHP community, so I don't see why this couldn't be added to this repo? You already do support `bower` and `npm`, so you do support package managers.

If anyone else is looking for Composer support for Leaflet, you might want to take a look at https://github.com/mrohnstock/leaflet

**Comment by blackice2999** [comment]
> we don't want to support every possible package manager

but you should support the most relevant package managers... the consequence are badly manages forked repositories this couldnt be your interest.

**Comment by tormi** [comment]
As a workaround, you can declare leaflet package by yourself in your project's `composer.json`, but that's not the simplest way:

```json
  "repositories": [
  {
    "type": "package",
    "package": {
      "name": "Leaflet/Leaflet",
      "version": "1.0.2",
      "dist": {
        "url": "https://github.com/Leaflet/Leaflet/archive/v1.0.2.zip",
        "type": "zip"
      }
    }
  }
]
```

**Pull request closed without merging**